uniflow/config.py

```diff
@@ -1,52 +1,14 @@
 """Flow config module."""
 
-from dataclasses import dataclass, field
-from typing import Dict
+from dataclasses import dataclass
 
-from uniflow.model.config import (
-    HuggingfaceModelConfig,
-    LMQGModelConfig,
-    ModelConfig,
-    OpenAIModelConfig,
-)
-from uniflow.schema import GuidedPrompt
+from uniflow.extract.config import ExtractConfig, ExtractTxtConfig
+from uniflow.transform.config import TransformConfig, TransformOpenAIConfig
 
 
 @dataclass
-class Config:
-    """Flow Config Class."""
+class PipelineConfig:
+    """Pipeline Config Class."""
 
-    flow_name: str = "ModelFlow"
-    guided_prompt_template: GuidedPrompt = GuidedPrompt()
-    num_thread: int = 1
-    model_config: ModelConfig = ModelConfig()
-
-
-@dataclass
-class OpenAIConfig:
-    """OpenAI Flow Config Class."""
-
-    flow_name: str = "OpenAIModelFlow"
-    guided_prompt_template: GuidedPrompt = GuidedPrompt()
-    num_thread: int = 1
-    model_config: ModelConfig = OpenAIModelConfig()
-
-
-@dataclass
-class HuggingfaceConfig:
-    """Huggingface Config Class."""
-
-    flow_name: str = "HuggingFaceModelFlow"
-    guided_prompt_template: GuidedPrompt = GuidedPrompt()
-    num_thread: int = 1
-    model_config: ModelConfig = HuggingfaceModelConfig()
-
-
-@dataclass
-class LMQGConfig:
-    """LMQG Config Class."""
-
-    flow_name: str = "LMQGModelFlow"
-    num_thread: int = 1
-    guided_prompt_template: Dict[str, str] = field(default_factory=lambda: {})
-    model_config: ModelConfig = LMQGModelConfig()
+    extract_config: ExtractConfig = ExtractTxtConfig()
```

Comment:
nit: it is a bad practice to put default value here.

Think from a user perspective. The must know what `extract_config` and `transform_config` to pick to setup the proper pipeline.

Pipeline should be generic interface for user to config.

You should remove `ExtractTxtConfig` and `TransformOpenAIConfig` and explain in your notebook when use PipelineConfig regarding why you pick these values.